packages/runtime/src/widgets/Charts/BarChart.tsx

```diff
@@ -27,23 +33,42 @@ const options: ChartOptions<"bar"> = {
   },
 };
 
-interface ChartDataSets {
-  label?: string;
-  data: number[];
-  backgroundColor?: string;
-  barPercentage?: number;
-  borderRadius?: number;
-}
+export const BarChart: React.FC<ChartProps> = (props) => {
+  const { id, styles, config } = props;
 
-export type BarChartProps = {
-  labels?: string[] | undefined;
-  datasets?: ChartDataSets[];
-  title?: Expression<string>;
-  [key: string]: unknown;
-} & EnsembleWidgetProps;
+  const [title, setTitle] = useState(config?.title);
+  const [labels, setLabels] = useState<string[]>(config?.labels || []);
+  const context = useScreenContext();
 
-export const BarChart: React.FC<BarChartProps> = (props) => {
-  const { labels, datasets, title, styles } = props;
+  const evaluatedDatasets = useMemo(() => {
+    return config?.datasets?.map((dataset) => ({
+      ...dataset,
+      data:
+        typeof dataset.data === "string"
+          ? evaluate(context as ScreenContextDefinition, dataset.data)
+          : dataset.data,
+
+      label: dataset.label?.startsWith("${")
+        ? (evaluate(
+            context as ScreenContextDefinition,
+            dataset.label,
+          ) as string)
+        : dataset.label,
+      backgroundColor:
+        typeof dataset.backgroundColor === "string" &&
+        dataset.backgroundColor.startsWith("${")
+          ? (evaluate(
+              context as ScreenContextDefinition,
+              dataset.backgroundColor,
+            ) as string | string[])
+          : dataset.backgroundColor,
+    }));
+  }, [config?.datasets]);
```

Comment:
You can pass in the entire string to the evaluate function. You may need to update the evaluate function's return logic to handle multiline object declarations.